instrumentation/action_pack/lib/opentelemetry/instrumentation/action_pack/patches/action_controller/metal.rb

```diff
@@ -16,6 +16,9 @@ def dispatch(name, request, response)
               rack_span.name = "#{self.class.name}##{name}" if rack_span.context.valid? && !request.env['action_dispatch.exception']
 
               add_rails_route(rack_span, request) if instrumentation_config[:enable_recognize_route]
+
+              rack_span.set_attribute('http.target', request.filtered_path)
```

Comment:
I think this is good to have and I agree we should honor it if set.

But, I'd like to try to avoid calling `set_attribute` on a hot path if it isn't necessary, as it's an expensive operation since it calls `mutex.synchronize` under the hood.

Looking into the internals, i think we can safely avoid calling `set_attribute` if either `request.query_string.empty?` (pseudocode but i think this is the right method), or the `action_dispatch.parameter_filter` header is not present? 
https://github.com/rails/rails/blob/f95c0b7e96eb36bc3efc0c5beffbb9e84ea664e4/actionpack/lib/action_dispatch/http/filter_parameters.rb#L65

wdyt?

Reply:
that sounds reasonable to me

Reply:
I've implemented this check using a `filtered_path != fullpath` check, which should give us what we need and shouldn't set the attribute if now filters have been applied.

Reply:
👍 tyty